git/gogit/transport.go

```diff
@@ -46,6 +46,20 @@ func transportAuth(opts *git.AuthOptions) (transport.AuthMethod, error) {
 		}
 		return nil, nil
 	case git.SSH:
+		// if the custom auth options don't provide a private key and known_hosts, we try
+		// to use the default known_hosts of the machine.
+		if len(opts.Identity)+len(opts.KnownHosts) == 0 && fallbackToDefaultKnownHosts {
+			authMethod, err := ssh.DefaultAuthBuilder(opts.Username)
+			if err != nil {
+				return nil, err
+			}
+			pkCallback, ok := authMethod.(*ssh.PublicKeysCallback)
+			if ok {
+				return &DefaultAuth{
+					pkCallack: pkCallback,
+				}, nil
+			}
```

Comment:
We may want to highlight the abnormal flow in which `authMethod` is not convertible to `ssh.PublicKeysCallback` by returning early with an error. Falling through to the previous code seems like an invalid execution path.

Reply:
i agree, thats why i had an `else` branch here which returned `nil, nil` which would lead to a host key error further down the stack, but i removed it at your suggestion during the previous review.

Reply:
The suggestion removed the redundant `else`, but kept the `return` statement.